### ISSUE TITLE
asl_f4_fw_postprocess does not create binary identical .ihex files across platforms

### ISSUE THREAD

**Issue by stieg** [autosportlabs/ASL_F4_bootloader#14]
In my attempts to debug the root cause of https://github.com/autosportlabs/RaceCapture-Pro_firmware/issues/753 I have learned that the f4_fw_postprocesser does not generate the same ihex code across platforms given the same .elf file.  This behavior is isolated to RCT builds; mk2 builds are not affected.  See the following output:

Jenkins:

```
[jenkins@ip-172-30-0-247 rct]$ python --version; sha256sum main.elf; asl_f4_fw_postprocess -f main.bin -e main.elf -o 0x08004000 -i main.ihex; sha256sum main.ihex
Python 2.7.5
2dfb53dd48b63d7511c0566039fd2c9ff52b136c5f54b754cf04d913c9c2c90d  main.elf
Found info block @ offset: 0x24c1c
Application CRC: 0x4fb122d9L
Info CRC: 0x69ba6a3bL
Application length: 169252
33dbcbfb85e55d9d8883f0321f6125dfbf60984b51f56b71ac61d460ad0dcb22  main.ihex
```

My Local Build Env:

```
[stieg@lucious rct]$ python --version; sha256sum main.elf; asl_f4_fw_postprocess -f main.bin -e main.elf -o 0x08004000 -i main.ihex; sha256sum main.ihex
Python 2.7.11
2dfb53dd48b63d7511c0566039fd2c9ff52b136c5f54b754cf04d913c9c2c90d  main.elf
Found info block @ offset: 0x24c1c
Application CRC: 0x4fb122d9L
Info CRC: 0x69ba6a3bL
Application length: 169252
3fb6003a03e75db7e8aea68676f72804177d9bde49dbec43cc87d65b0bf937c1  main.ihex
```

Not sure if its something related to the platform or the python version, but it does cause the bootloader on RCT to get corrupted.  The top build causes the bootloader region to be smashed while the bottom build works just fine.

**Comment by stieg** [comment]
Uploaded the two .ihex files that have this issue.  The former will kill your bootloader if you flash it using the asl_f4_flasher tool.

[murderous.zip](https://github.com/autosportlabs/ASL_F4_bootloader/files/376953/murderous.zip)
[working.zip](https://github.com/autosportlabs/ASL_F4_bootloader/files/376954/working.zip)

**Comment by brentpicasso** [comment]
I installed the latest here from master and got this: (edit) same as your machine:

python --version; sha256sum main.elf; asl_f4_fw_postprocess -f main.bin -e main.elf -o 0x08004000 -i main.ihex; sha256sum main.ihex
Python 2.7.6
2dfb53dd48b63d7511c0566039fd2c9ff52b136c5f54b754cf04d913c9c2c90d  main.elf
Found info block @ offset: 0x24c1c
Application CRC: 0x4fb122d9L
Info CRC: 0x69ba6a3bL
Application length: 169252
3fb6003a03e75db7e8aea68676f72804177d9bde49dbec43cc87d65b0bf937c1  main.ihex
brent@brent-desktop:~/git-projects/RaceCapture-Pro_firmware/platform/rct$

**Comment by brentpicasso** [comment]
^^ corrected. Looks like my machine makes the same as yours.

**Comment by brentpicasso** [comment]
And, firmware flashed from my image boots correctly.

**Comment by Jeff-Ciesielski** [comment]
Just had a call with @stieg: I think that the issue is that jenkins needs an updated copy of ihextools.  There was a bug in the old one which caused incorrect offsets to be calculated.

**Comment by stieg** [comment]
Ok just upgraded the build machine.  Here is the output again:

```
[jenkins@ip-172-30-0-247 rct]$ cat $(which asl_f4_fw_postprocess)
#!/usr/bin/python
# EASY-INSTALL-ENTRY-SCRIPT: 'asl-f4-loader==0.1.2','console_scripts','asl_f4_fw_postprocess'
__requires__ = 'asl-f4-loader==0.1.2'
import sys
from pkg_resources import load_entry_point

if __name__ == '__main__':
    sys.exit(
        load_entry_point('asl-f4-loader==0.1.2', 'console_scripts', 'asl_f4_fw_postprocess')()
    )
[jenkins@ip-172-30-0-247 rct]$ python --version; sha256sum main.elf; asl_f4_fw_postprocess -f main.bin -e main.elf -o 0x08004000 -i main.ihex; sha256sum main.ihex
Python 2.7.5
2dfb53dd48b63d7511c0566039fd2c9ff52b136c5f54b754cf04d913c9c2c90d  main.elf
Found info block @ offset: 0x24c1c
Application CRC: 0x4fb122d9L
Info CRC: 0x69ba6a3bL
Application length: 169252
33dbcbfb85e55d9d8883f0321f6125dfbf60984b51f56b71ac61d460ad0dcb22  main.ihex
```

Unfortunately it still produces the poisonous ihex file it would seem

**Comment by brentpicasso** [comment]
is there a way to verify the version of ihextools it's using?

I noticed that when I upgraded the Lab computer, I didn't upgrade ihextools, yet asl_f4_loader installed without complaint (sudo python setup.py install)

Will test updating ihextools at lab at next opportunity

**Comment by stieg** [comment]
Issue was related to ihextools dependency being behind.  Needed 1.1.0 of ihextools for it to work properly yet dependency was at 1.0.0 which had the bug.

**Comment by stieg** [comment]
Issue will be resolved by version 0.2.0 which will have the fixes for this issue along with a couple others.
